### PR TITLE
Use test-context for DeploymentsComponent spec

### PR DESCRIPTION
**Pull request by andrewazores**
Refactor DeploymentsComponent spec to use TestContext. This is cleaner with less repeat boilerplate, but more importantly, hopefully will eliminate an intermittent test failure as seen here: https://jenkins.cd.test.fabric8.io/job/fabric8-ui/job/fabric8-ui/job/PR-2305/4/display/redirect (due to better test component set up performed by the TestContext).

### CODE REVIEW

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2311-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2311-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/deployments.component.spec.ts [hunk 1]
```diff
@@ -40,69 +36,55 @@ class FakeDeploymentAppsComponent {
...
-      ]
-    });
-
-    fixture = TestBed.createComponent(DeploymentsComponent);
-    component = fixture.componentInstance;
+  beforeAll(() => {
+    mockSvc.getApplications.and.returnValue(mockApplications);
+    mockSvc.getEnvironments.and.returnValue(mockEnvironments);
+    mockSvc.getCpuStat.and.returnValue(Observable.of({ used: 1, total: 2 }));
+    mockSvc.getMemoryStat.and.returnValue(Observable.of({ used: 1, total: 2 }));
```
Comment (jiekang) [inline src/app/space/create/deployments/deployments.component.spec.ts]: Are CpuStat and MemoryStat used in any tests here? If not used, I'd rather they throw unimplemented. It follows with other locations that create this mock service. It then becomes clear that this component doesn't rely on those service functions.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2311-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2311-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2311-9 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2311-fabric8-ui.badger.fabric8.io

**Pull request merged**